facade-app/src/test/LeftNav.test.tsx

```diff
@@ -0,0 +1,254 @@
+        ['VulnerableApp.UnionBasedSQLInjectionVulnerability.LEVEL_3', 'LEVEL_3'],
+        ['VulnerableApp.UnionBasedSQLInjectionVulnerability.LEVEL_4', 'LEVEL_4'],
+        ['VulnerableApp.SSRFVulnerability.LEVEL_1', 'LEVEL_1'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_1', 'LEVEL_1'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_2', 'LEVEL_2'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_3', 'LEVEL_3'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_4', 'LEVEL_4'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_5', 'LEVEL_5'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_6', 'LEVEL_6'],
+        ['VulnerableApp.PersistentXSSInHTMLTagVulnerability.LEVEL_7', 'LEVEL_7'],
+        ['VulnerableApp.XSSWithHtmlTagInjection.LEVEL_1', 'LEVEL_1'],
+        ['VulnerableApp.XSSWithHtmlTagInjection.LEVEL_2', 'LEVEL_2'],
+        ['VulnerableApp.XSSWithHtmlTagInjection.LEVEL_3', 'LEVEL_3'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_1', 'LEVEL_1'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_2', 'LEVEL_2'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_3', 'LEVEL_3'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_4', 'LEVEL_4'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_5', 'LEVEL_5'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_6', 'LEVEL_6'],
+        ['VulnerableApp.XSSInImgTagAttribute.LEVEL_7', 'LEVEL_7'],
+        ['VulnerableApp.XXEVulnerability.LEVEL_1', 'LEVEL_1'],
+        ['VulnerableApp.XXEVulnerability.LEVEL_2', 'LEVEL_2'],
+        ['VulnerableApp.XXEVulnerability.LEVEL_3', 'LEVEL_3'],
+        ['VulnerableApp.XXEVulnerability.LEVEL_4', 'LEVEL_4'],
+    ]
+
+    it.each(levels)('should have test id %s and text %s', (testId, label) => {
+        const mock = () => jest.fn();
+        render(<LeftNav globalState={testFixture} setGlobalState={mock} />)
+
+        const link = screen.getByTestId(testId);
+        expect(link).toBeInTheDocument();
+        expect(link).toHaveTextContent(label);
+    })
+  });
+
+  describe("VulnerableApp-jsp", () => {
+      it("renders correctly", () => {
+          const mock = () => jest.fn();
+          render(<LeftNav globalState={testFixture} setGlobalState={mock}/>)
+
+          const item = screen.getByText("VulnerableApp-jsp")
+          const serverIcon = screen.getAllByRole("img")[1]
+
+          expect(item).toBeInTheDocument();
+          expect(serverIcon).toBeInTheDocument();
+      })
+
+      const inputs = [
+          ["FileUpload"]
+      ]
+
+      it.each(inputs)(`should have sub item %s`, (text) => {
+          const mock = () => jest.fn();
+          render(<LeftNav globalState={testFixture} setGlobalState={mock} />);
+
+          const item = screen.getAllByText(text)[0]
+          expect(item).toBeInTheDocument()
+      })
+
+      const levels = [
+          ['VulnerableApp-jsp.FileUpload.LEVEL_1', 'LEVEL_1'],
```

Comment:
Same for this one as well. This is only available if VulnerableApp-jsp is integrated so let's not have check on these values.

Reply:
or is it mocked data from fixtures.ts?

Reply:
these are mocked from fixtures.ts too